src/NoWrapTypography/NoWrapTypography.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import { Tooltip, Typography } from "@mui/material";
+
+import PropTypes from "prop-types";
+
+/**
+ * Typography component to show a tooltip if the text overflows.
+ */
+export default function NoWrapTypography({ children, sx }) {
+  const [tooltipEnabled, setTooltipEnabled] = useState(false);
+
+  // if the text overflows its bounding box, then show the tooltip
+  const handleShouldShow = ({ currentTarget }) => {
+    if (currentTarget.scrollWidth > currentTarget.clientWidth) {
+      setTooltipEnabled(true);
+    }
+  };
+
+  const hideTooltip = () => setTooltipEnabled(true);
+
+  return (
+    <Tooltip
+      title={children}
+      disableHoverListener={!tooltipEnabled}
+      onMouseEnter={handleShouldShow}
+      onMouseLeave={hideTooltip}
+    >
+      <Typography
+        noWrap
+        sx={sx}
```

Comment:
Why style and sx?  why don't you pass it just like 
overflow="hidden"
        textOverflow="ellipsis"
        whiteSpace="nowrap" ?

Reply:
Typography doesn't allow overflow, text-overflow, and white space as props.
https://mui.com/material-ui/api/typography/

Reply:
I don't think so. It's been used here and working https://github.com/IPG-Automotive-UK/test-center/blob/main/app/src/components/TypographyWithOverflowTooltip.tsx#L29

Reply:
> Typography doesn't allow overflow, text-overflow, and white space as props. https://mui.com/material-ui/api/typography/

https://mui.com/system/properties/

Reply:
As a CSS utility, the Typography component also supports all [system](https://mui.com/system/properties/) properties. You can use them as props directly on the component.